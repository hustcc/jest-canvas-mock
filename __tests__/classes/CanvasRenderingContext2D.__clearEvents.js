let ctx;
beforeEach(() => {
  // get a new context each test
  ctx = document.createElement('canvas').getContext('2d');
});

afterEach(() => {
  const events = ctx.__getEvents();
  expect(events).toMatchSnapshot();
});

describe('__clearEvents', () => {
  it('should clear the list of events', () => {
    ctx.fillRect(1, 2, 3, 4);
    ctx.__clearEvents();
  });

  it('should not prevent additional events from being collected', () => {
    ctx.fillRect(1, 2, 3, 4);
    ctx.__clearEvents();
    ctx.fillRect(1, 2, 3, 4);
  });
});
