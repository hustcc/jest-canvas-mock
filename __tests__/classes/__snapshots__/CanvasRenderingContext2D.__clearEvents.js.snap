// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__clearEvents should clear the list of events 1`] = `Array []`;

exports[`__clearEvents should not prevent additional events from being collected 1`] = `
Array [
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "fillRect",
  },
]
`;
